src/libertem/io/dataset/seq.py

```diff
@@ -269,32 +441,53 @@ def _maybe_load_mrc(self, path):
         data_dict = mrcReader(path)
         return np.squeeze(data_dict['data'])
 
+    def _load_xml_from_file(self, path):
+        if not os.path.exists(path + ".Config.Metadata.xml"):
+            return None
+        if not os.path.exists(path + ".metadata"):
+            return None
+        else:
+            tree = ET.parse(path + ".Config.Metadata.xml")
+            root = tree.getroot()
+            file = open(path + ".metadata", mode="rb")
+            met = file.read()
```

Comment:
When reading from files, the context manager form should be preferred to properly close the file in all cases:

```suggestion
            with open(path + ".metadata", mode="rb") as file:
                met = file.read()
```

(this is especially important for some Python implementations that don't immediately GC objects once they go out of scope, like PyPy)